core/org.wso2.carbon.ndatasource.rdbms/src/main/java/org/wso2/carbon/ndatasource/rdbms/ConfigurableCorrelationLogService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.ndatasource.rdbms;
+
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.logging.correlation.CorrelationLogConfigurable;
+import org.wso2.carbon.logging.correlation.bean.ImmutableCorrelationLogConfig;
+
+@Component(
+        immediate = true,
+        service = CorrelationLogConfigurable.class
+)
+public class ConfigurableCorrelationLogService implements CorrelationLogConfigurable {
+    private static boolean enable;
+    private static String[] deniedThreads;
+
+    @Override
+    public String getName() {
+        return "jdbc";
+    }
+
+    @Override
+    public void onConfigure(ImmutableCorrelationLogConfig config) {
+        enable = config.isEnable();
+        deniedThreads = config.getDeniedThreads();
+    }
+
+    public static boolean isEnable() {
+        return enable;
+    }
+
+    public static String[] getDeniedThreads() {
+        return deniedThreads;
+    }
```

Comment:
I am not sure it is the best way but how about using @Activate method here and passing the reference "this" to a ServiceReferenceHolder. Using that, we can get the service instance and check for isEnable/getDeniedThreads without using static methods.

@lasanthaS @tharindu1st